kairon/api/app/routers/history.py

```diff
@@ -308,81 +360,101 @@ async def word_cloud(
-async def user_intent_dropoff(month: int = Query(default=1, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def user_intent_dropoff(
+        from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+        to_date: datetime.date = datetime.date.today(),
+        current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the identified intents and their counts for users before dropping off from the conversations.
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/metrics/intents/dropoff',
-        {'month': month}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y")}
     )
 
 
 @router.get("/metrics/user/sessions/unsuccessful", response_model=Response)
-async def unsuccessful_session_count(month: int = Query(default=1, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def unsuccessful_session_count(
+        from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+        to_date: datetime.date = datetime.date.today(),
+        current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the count of sessions that encountered a fallback for a particular user.
     """
     fallback_action, nlu_fallback_action = DataUtility.load_fallback_actions(current_user.get_bot())
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/metrics/sessions/unsuccessful',
-        {'month': month, 'action_fallback': fallback_action, 'nlu_fallback': nlu_fallback_action}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y"),
+         'action_fallback': fallback_action, 'nlu_fallback': nlu_fallback_action}
     )
 
 
 @router.get("/metrics/user/sessions/total", response_model=Response)
-async def total_sessions(month: int = Query(default=1, ge=1, le=6), current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
+async def total_sessions(
+        from_date: datetime.date = datetime.date.today() - datetime.timedelta(30),
+        to_date: datetime.date = datetime.date.today(),
+        current_user: User = Security(Authentication.get_current_user_and_bot, scopes=TESTER_ACCESS)):
     """
     Fetches the total session count for users for the past months.
     """
     return Utility.trigger_history_server_request(
         current_user.get_bot(),
         f'/api/history/{current_user.get_bot()}/metrics/sessions/total',
-        {'month': month}
+        {'from_date': from_date.strftime("%d-%m-%Y"), 'to_date': to_date.strftime("%d-%m-%Y")}
```

Comment:
make a separate utility to convert date to string

Reply:
I have written separate utility function and made changes accordingly